### PR TITLE
Finance Debt Claim Page

### LINKED ISSUE #141: Finance debt and claim component Page
Page Url: /transaction/ (with different params for both page)
Module: Finance - claim & debt
Authority:
- CC Admin

Usage: View claim and debt

- [x] Development

### LINKED ISSUE #143: Custom tab in some files will not update when changing language
Current Behaviour: Custom tab in some files will not update when changing language
Expected Behaviour: Custom tab should change language as well
Steps To Reproduce?: 
1. In Member Debt page, change language to Chinese
2. Change back to English, this time the language of custom tab wont be changed
Screenshots of UI?:
Screenshot_853
Additional Details: Adding key props with current timestamp in custom tab that is not in eventTabList

### PR DESCRIPTION
**Pull request by CarlosSia**
resolve #141 
resolve #143 

### UI Screenshot
**Debt Page**
Screenshot_845
Screenshot_856
Screenshot_857
Screenshot_848
Screenshot_858
Screenshot_859

**Claim Page**
Screenshot_868
Screenshot_847
Screenshot_846
Screenshot_849
Screenshot_850
Screenshot_852

### CODE REVIEW

**Review by HowardYaw — COMMENTED** [review]
@CarlosSia  Why the CustomTab need to pass in key with Date.now()?

**Comment by CarlosSia** [comment]
> @CarlosSia Why the CustomTab need to pass in key with Date.now()?

@HowardYaw It is to resolve issue #143 , because when the user keep changing the language, after the second time React will think that they already update the Custom Tab cause the Custom Tab language will remain the same. So to force React always updating the Custom Tab need to use unique prop to make React to think the Custom Tab is different now and will always update it.

reference link:
https://stackoverflow.com/questions/38892672/react-why-child-component-doesnt-update-when-prop-changes

**Comment by HowardYaw** [comment]
@CarlosSia but previously Event Module Page do update after language changed
without the Date.now()

**Comment by CarlosSia** [comment]
> @CarlosSia but previously Event Module Page do update after language changed
> without the Date.now()

@HowardYaw Ya, because the event module page need to pass in the eventId from the page, so every time the language change will cause the page to pass the eventId into CustomTab and make it to update. But for other page even the event main tab page that show previous and current event they wont update when change the language

**Comment by HowardYaw** [comment]
@CarlosSia  why have new changes again?
I thought you are very confident that no need changes on Saturday Meeting

**Review by HowardYaw — APPROVED** [review]

File: src/main/webapp/app/shared/components/finance-component-listing-card/finance-component-listing-card.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+              onClick={completedActionCallback}
+            />
+          </div>
+          <div className="mt-2">
+            <FinanceActionButton
+              name={<Translate contentKey={`entity.action.${invalidAction}`}>{invalidAction}</Translate>}
+              color={invalidActionColor}
+              onClick={invalidActionCallback}
+            />
+          </div>
```
Comment (junzai97) [inline src/main/webapp/app/shared/components/finance-component-listing-card/finance-component-listing-card.tsx:49]: image

It seem like there will be overflow when if the screen size is too small

**Review by LUXIANZE — APPROVED** [review]

File: src/main/webapp/app/shared/components/finance-component-listing-card/finance-component-listing-card.tsx (hunk 1 above)
Comment (HowardYaw) [reply src/main/webapp/app/shared/components/finance-component-listing-card/finance-component-listing-card.tsx:49]: Issue Created https://github.com/Five-Fishes/Club-Management-React-Client/issues/150

**Pull request merged**